src/client/api/tests/job_tests.c

```diff
@@ -0,0 +1,221 @@
+		return getenv_jobid_env_return;
+	} else if (getenv_jobid_env_return &&
+		strncmp(name, getenv_jobid_env_return, MAX_ENV_NAME) == 0) {
+		return getenv_jobid_return;
+	} else {
+		return NULL;
+	}
+}
+
+static pid_t getpid_pid;
+pid_t getpid(void)
+{
+	return getpid_pid;
+}
+
+static char *uname_nodename;
+int uname (struct utsname *buf)
+{
+	if (!buf) {
+		errno = EFAULT;
+		return -1;
+	}
+
+	strncpy(buf->nodename, uname_nodename, _UTSNAME_LENGTH);
+	return 0;
+}
+
+/*
+ * Unit test setup and teardown
+ */
+
+static int
+setup_job_mocks(void **state)
+{
+	/* Initialize mock values to something sane */
+	getenv_daos_jobid_return = NULL;
+	getenv_jobid_env_return = NULL;
+	getenv_jobid_return = NULL;
+	getpid_pid = 0;
+	uname_nodename = NULL;
+	return 0;
+}
+
+static int
+teardown_job_mocks(void **state)
+{
+	return 0;
+}
+
+static int
+craft_jobid(char **jobid, char *nodename , pid_t pid)
+{
+	int   ret = 0;
+	char *tmp_jobid;
+
+	if (!jobid)
+		return -1;
+
+	ret = asprintf(&tmp_jobid, "%s-%d", nodename, pid);
+	if (ret == -1)
+		return -1;
+
+	*jobid = tmp_jobid;
+	return 0;
+}
+
+/*
+ * Client lib agent function tests
+ */
+static void
+test_dc_job_init_no_env(void **state)
+{
+	int	 ret = 0;
+	char	*default_jobid = NULL;
+
+	uname_nodename = "testhost";
+	getpid_pid = 1000;
+
+	ret = craft_jobid(&default_jobid, uname_nodename, getpid_pid);
+	assert_return_code(ret, 0);
+
+	dc_job_init();
+	/* Make sure we checked the right environment variable */
+	assert_string_equal(dc_jobid_env, DEFAULT_JOBID_ENV);
+
+	/* Make sure we crafted a default jobid  */
+	assert_string_equal(dc_jobid, default_jobid);
+
+	dc_job_fini();
+}
+
+static void
+test_dc_job_init_with_jobid(void **state)
+{
+	getenv_daos_jobid_return = "test-jobid";
+
+	dc_job_init();
+	/* Make sure we checked the right environment variable */
+	assert_string_equal(dc_jobid_env, DEFAULT_JOBID_ENV);
+
+	/* Make sure we crafted a default jobid  */
+	assert_string_equal(dc_jobid, getenv_daos_jobid_return);
+
+	dc_job_fini();
+}
+
+static void
+test_dc_job_init_with_jobid_env(void **state)
+{
+	int	 ret = 0;
+	char	*default_jobid = NULL;
+
+	uname_nodename = "testhost";
+	getpid_pid = 1000;
+	getenv_jobid_env_return = "other-jobid-env";
+
+	ret = craft_jobid(&default_jobid, uname_nodename, getpid_pid);
+	assert_return_code(ret, 0);
+
+	dc_job_init();
+	/* Make sure we checked the right environment variable */
+	assert_string_equal(dc_jobid_env, getenv_jobid_env_return);
+
+	/* Make sure we crafted a default jobid  */
+	assert_string_equal(dc_jobid, default_jobid);
+
+	dc_job_fini();
+}
+
+static void
+test_dc_job_init_with_jobid_env_and_jobid(void **state)
+{
+	getenv_jobid_env_return = "other-jobid-env";
+	getenv_jobid_return = "test-jobid";
+
+	dc_job_init();
+	/* Make sure we checked the right environment variable */
+	assert_string_equal(dc_jobid_env, getenv_jobid_env_return);
+
+	/* Make sure we crafted a default jobid  */
+	assert_string_equal(dc_jobid, getenv_jobid_return);
```

Comment:
Comment isn't correct, we got the job ID from the env variable.

Reply:
fixed.